src/orion/storage/base.py

```diff
@@ -104,6 +115,10 @@ def retrieve_result(self, trial, *args, **kwargs):
         """
         raise NotImplementedError()
 
+    def fetch_pending_trials(self, experiment):
+        """Fetch trials that have not run yet"""
```

Comment:
This should be all trials that can be executed. It includes interrupted and suspended trials, which mean they were executed yet but didn't completed.